From cc214c1d33fdff654f4a8b1b2142847300fbcbce Mon Sep 17 00:00:00 2001
From: Christian Lamparter <chunkeey@googlemail.com>
Date: Fri, 5 Apr 2013 00:20:55 +0200
Subject: [PATCH 10/65] r92su: fix early rx aggregation timeout

Due to a off-by-one bug, the code did not wait
for the HT peer to resent a missing part of an
AMPDU.

Signed-off-by: Christian Lamparter <chunkeey@googlemail.com>
---
 r92su/rx.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

--- a/drivers/net/wireless/r92su/rx.c
+++ b/drivers/net/wireless/r92su/rx.c
@@ -805,7 +805,7 @@ r92su_sta_reorder_release(struct r92su *
 			}
 
 			if (skipped &&
-			    time_is_before_jiffies(tid->reorder_time[j]))
+			    time_is_after_jiffies(tid->reorder_time[j]))
 				goto set_release_timer;
 
 			tid->head_seq = ieee80211_sn_add(tid->head_seq,
@@ -836,8 +836,6 @@ set_release_timer:
 	}
 }
 
-#define HT_RX_REORDER_INIT (HZ * 5)
-
 static enum r92su_rx_control_t
 r92su_rx_ampdu_reorder(struct r92su *r92su, struct sk_buff *skb,
 		       struct r92su_bss_priv *bss_priv,
@@ -898,8 +896,6 @@ r92su_rx_ampdu_reorder(struct r92su *r92
 		tid->len++;
 		r92su_sta_reorder_release(r92su, tid, queue);
 	}
-
-	tid->last_rx = jiffies + HT_RX_REORDER_INIT;
 	spin_unlock(&tid->lock);
 
 out:
